src/market/market_event_utils.cairo

```diff
@@ -10,3 +10,19 @@ fn emit_market_pool_value_info(
     market_tokens_supply: u128
 ) {}
 
+fn emit_virtual_swap_inventory_updated(
+    event_emitter: IEventEmitterDispatcher,
+    market: ContractAddress,
+    is_long_token: bool,
+    virtual_market_id: felt252,
+    delta: i128,
+    next_value: u128
+) {}
```

Comment:
why those functions are empty ?

Reply:
I call those functions in `apply_delta_to_virtual_inventory_for_swaps()` and `apply_delta_to_virtual_inventory_for_positions()`

Reply:
But those functions are in the `MarketEventUtils.sol` file in the gmx repository

Reply:
yes but in cairo you need to be in a smart contract (have a `self` variable) to emit events, which is not the case here

see
https://github.com/delaaxe/satoru/blob/main/src/market/market_utils.cairo#L285
vs
https://github.com/gmx-io/gmx-synthetics/blob/77a2ff39f1414a105e8589d622bdb09ac3dd97d8/contracts/market/MarketUtils.sol#L499

Reply:
ok thank you very much